src/ui/Analytics/Analytics.ts

```diff
@@ -35,96 +35,16 @@ export interface IAnalyticsOptions {
- * Framework offers helper functions to communicate with the Coveo Usage Analytics REST API, so you do not have to write
- * code to call the API directly.
- *
- * ## Create a Custom Event Cause and Metadata
- *
- * ```
- * // Each event should have a unique "name" attribute (e.g., "searchboxSubmit", "searchboxClear").
- * // The "type" attribute allows you to group similar event types when doing reporting (e.g., "searchbox").
- * var customEventCause = {name: 'customEventName', type: 'customEventType'};
- *
- * // This is arbitrary data you want to send to the Coveo Usage Analytics in order to create custom dimensions (see
- * // [Creating and Managing Dimensions on Custom Metada](http://www.coveo.com/go?dest=cloudhelp&lcid=9&context=142)).
- * var metadata = {key1: 'value1', key2: 'value2'};
- * ```
- *
- * See {@link IAnalyticsActionCause}
- *
- * ## Send a Custom Event
- * ```
- * Coveo.logCustomEvent(document.querySelector('#search'), customEventCause, metadata);
- * ```
- *
- * Or, using the jQuery extension:
- * ```
- * $('#search').coveo('logCustomEvent', customEventCause, metadata);
- * ```
- * See {@link logCustomEvent}.
- *
- * ## Send a Custom Search Event
- *
- * NB: If you want to log a search event, be sure to always call the `logSearchEvent` helper before you call
- * `executeQuery`.
- *
- * ```
- * function myCustomButtonWasClicked() {
- *   Coveo.logSearchEvent(document.querySelector('#search'), customEventCause, metadata);
- *   Coveo.executeQuery(document.querySelector('#search'));
- * ```
- *
- * Or, using the jQuery extension:
- * ```
- * function myCustomButtonWasClicked() {
- *   $('#search').coveo('logSearchEvent', customEventCause, metadata);
- *   $('#search').coveo('executeQuery');
- * }
- * ```
- * See {@link logSearchEvent}.
- *
- * ## Send a Custom Search-As-You-Type Event
- *
- * NB: If you want to log a search-as-you-type event, be sure to always call the `logSearchAsYouTypeEvent` helper before
- * you call `executeQuery`.
- *
- * ```
- * function myCustomButtonWasClicked() {
- *     Coveo.logSearchAsYouTypeEvent(document.querySelector('#search'), customEventCause, metadata);
- *     Coveo.executeQuery(document.querySelector('#search'));
- * ```
- *
- * Or, using the jQuery extension:
- * ```
- * function myCustomButtonWasClicked() {
- *   $('#search').coveo('logSearchAsYouTypeEvent', customEventCause, metadata);
- *   $('#search').coveo('executeQuery');
- * }
- * ```
- * See {@link logSearchAsYouTypeEvent}.
- *
- * ## Send a Custom Click Event
- * ```
- * Coveo.logClickEvent(document.querySelector('#search'), customEventCause, metadata, result);
- * ```
- *
- * Or, using the jQuery extension:
- * ```
- * $('#search').coveo('logClickEvent', customEventCause, metadata, result);
- * ```
- * See {@link logClickEvent}.
+ * See also [Sending Custom Analytics Events](https://developers.coveo.com/x/KoGfAQ) for more advanced use cases.
```

Comment:
I added / changed a lot of things in this topic, you might want to check it out :)

https://developers.coveo.com/x/KoGfAQ